tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/paging/PagingSendTest.java

```diff
@@ -288,6 +292,149 @@ public void testPagingDoesNotDuplicateBatchMessagesAfterPagingStarted() throws E
+         SimpleString queueAddr = SimpleString.of(queueName);
+         session.createQueue(QueueConfiguration.of(queueAddr));
+
+         int size = 1024 * 50;
+         AddressSettings addressSettings = new AddressSettings();
+         addressSettings.setPageFullMessagePolicy(PageFullMessagePolicy.FAIL);
+         addressSettings.setPageSizeBytes(size);
+         addressSettings.setPageLimitBytes(Long.valueOf(size * 10));
+         addressSettings.setMaxSizeBytes(size);
+
+         server.getAddressSettingsRepository().addMatch(queueName, addressSettings);
+
+         int totalMessages = 30;
+         int messageSize = 1024 * 10;
+         sendMessageBatch(totalMessages, messageSize, session, queueAddr);
+
+         Queue queue = server.locateQueue(queueAddr);
+         assertTrue(waitForMessages(queue, totalMessages, 10000));
+
+         PagingStore queuePagingStore = queue.getPagingStore();
+         assertTrue(queuePagingStore != null && queuePagingStore.isPaging());
+
+         long existingPages = queuePagingStore.getNumberOfPages();
+         assertTrue(existingPages > 4);
+
+         // restart the server with a smaller pageLimitSize < existing pages.
+         server.stop(true);
+         waitForServerToStop(server);
+
+         addressSettings.setPageLimitBytes(Long.valueOf(size * 4));
+         server.getAddressSettingsRepository().addMatch(queueName, addressSettings);
+
+         // server will start regardless current pages > pageLimitBytes
+         try {
+            server.start();
+            waitForServerToStart(server);
+
+            // verify current situation
+            queue = server.locateQueue(queueAddr);
+
+            assertTrue(waitForMessages(queue, totalMessages, 10000));
+
+            queuePagingStore = queue.getPagingStore();
+            assertTrue(queuePagingStore != null && queuePagingStore.isPaging() && queuePagingStore.isPageFull());
+
+            long currentPages = queuePagingStore.getNumberOfPages();
+            assertEquals(existingPages, currentPages);
+
+            long maxPages = queuePagingStore.getPageLimitBytes() / queuePagingStore.getPageSizeBytes();
+            assertTrue(currentPages > maxPages);
+
+            //consume messages until current pages goes down to below maxPage
+            locator = createFactory(isNetty());
+            final int numMessages = 25;
+            try (ClientSessionFactory csf = createSessionFactory(locator)) {
+               session = csf.createSession(false, true);
+               session.start();
+               ClientConsumer consumer = session.createConsumer(queueName);
+               for (int i = 0; i < numMessages; i++) {
+                  ClientMessage message = consumer.receive(5000);
+                  assertNotNull(message);
+                  message.acknowledge();
+                  session.commit();
+               }
+
+               currentPages = queuePagingStore.getNumberOfPages();
+               assertTrue(currentPages < maxPages);
```

Comment:
Should it asssert that that paging store is _not_ pageFull, rather than assuming it isnt? The below bits could fail on the first send attempt and erroneously pass since they never actually check the store.